### ISSUE TITLE
cmus 2.7.0 doesn't recognize compilations

### ISSUE THREAD

**Issue by audriusk** [cmus/cmus#302]
After upgrade from 2.6.0 to 2.7.0 all my compilations aren't recognized as such. Deleting .cmus directory and letting cmus create a new one didn't make any difference. My music collection consists of FLAC files only, all compilations have `compilation` tag set to 1, the tag was added using Ex Falso tag editor from Quod Libet music player.

**Comment by flyingmutant** [comment]
I don't see any changes to compilation detection in 2.6.0...2.7.0 timeframe in `git log`, apart from cb7eda1fb2da51158ccf9044da345469c25216c2, which is harmless for sure.

Can you please upload some test file, so I can have a look?

**Comment by audriusk** [comment]
My music is legally purchased so I wouldn't really want to upload it anywhere public. But the cb7eda1 commit that you noted led me to investigate `comment.c` file more thoroughly, and here's the diff which I came with and which fixes my problem:

```diff
--- a/comment.c
+++ b/comment.c
@@ -56,8 +56,8 @@ int track_is_va_compilation(const struct keyval *comments)
        const char *c, *aa;

        aa = comments_get_albumartist(comments);
-       if (aa)
-               return is_various_artists(aa);
+       if (aa && is_various_artists(aa))
+               return 1;

        c = keyvals_get_val(comments, "compilation");

```

The problem was that `comments_get_albumartist()` function returns either value of `albumartist` or `artist`, in my case `albumartist` was missing, but `artist` was always set, thus the following conditional statement was always returning from function and skipping the `compilation` check below it.

Please check if this fix is correct, as this is my first foray into C code base after a very long time (I usually spend my time in Python code).

**Comment by flyingmutant** [comment]
You are absolutely correct — the commit in question is in fact wrong and harmful. I've reverted it for now; please try to use cmus' master branch.

I'll try to release a maintenance release with the fix included in the near future.

**Comment by audriusk** [comment]
I can confirm that building from master branch handles compilations correctly now, thanks!

**Comment by flyingmutant** [comment]
Can you please post full tags of some of your tracks?

**Comment by flyingmutant** [comment]
Can you please tell what do you mean by «all my compilations aren't recognized as such» in the initial message?

Thinking about this issue some more, I am not sure I can understand what the bug was in the first place.
If you compilation tracks do have compilation tag set, and do have an artist set — they should appear under this artist in the tree view, not «Various Artists». «Various Artists» is for cases where we *can not* attribute the album to some one artist; with artist tag being set we definitely can (and should).

Maybe I am a bit tired and not thinking properly right now, but cb7eda1 looks, in fact, completely OK to me now.

**Comment by audriusk** [comment]
Sorry for not being clear. What I meant was that compilation tracks, which before the upgrade to 2.7.0 were placed under "Various Artists", now were scattered all over the entire music collection each belonging to its own artist and preserving the same album name. Here's an example of collection consisting of two tracks only:

```sh
$ for f in *.flac ; do printf "\n$f\n" ; metaflac --export-tags-to=- $f ; done

01_dopplereffekt_-_delta_wave.flac
album=Hypnagogia
artist=Dopplereffekt
compilation=1
title=Delta Wave
date=2014
tracknumber=1

02_objekt_-_ganzfeld.flac
album=Hypnagogia
artist=Objekt
compilation=1
title=Ganzfeld
date=2014
tracknumber=2
```

Before upgrade these tracks were under "Various Artists", on 2.7.0 they were displayed as two separate albums under each artist, consisting of single track.

**Comment by flyingmutant** [comment]
I see — thanks a lot!

**Comment by flyingmutant** [comment]
Can you please try out https://github.com/flyingmutant/cmus/commit/b5ee1b47810e373c44ae33a8b2d9235ccd4cfc0f?

**Comment by audriusk** [comment]
Just tried that commit with fresh cmus config -- working great!

**Comment by flyingmutant** [comment]
I've recently released 2.7.1 which does contain the fix.
